src/map/atcommand.c

```diff
@@ -5160,17 +5160,17 @@ ACMD(storeall)
 
 ACMD(clearstorage)
 {
-	int i, j;
+	int i;
 
 	if (sd->state.storage_flag == STORAGE_FLAG_NORMAL) {
 		clif->message(fd, msg_fd(fd,250));
 		return false;
 	}
 
-	j = sd->status.storage.storage_amount;
-	for (i = 0; i < j; ++i) {
-		storage->delitem(sd, i, sd->status.storage.items[i].amount);
+	for (i = 0; i < VECTOR_LENGTH(sd->storage); ++i) {
```

Comment:
This won't work properly: it's iterating on the same set it's deleting (the indexes as well as VECTOR_LENGTH will change during the loop).

If you really want to do this, and you really want to use vectors, then you have to loop in reverse:

```
while (VECTOR_LENGTH(sd->storage) > 0) {
    storage->delitem(sd, VECTOR_LENGTH(sd->storage) - 1, VECTOR_LAST(sd->storage).amount);
}
```

This will also be more efficient (deleting the last element of a vector doesn't cause it to have to move every other element multiple times)